src/library/basesqltablemodel.cpp

```diff
@@ -280,13 +280,71 @@ void BaseSqlTableModel::select() {
         qDebug() << "Rows actually received:" << rowInfos.size();
     }
 
+    if (!m_sortedTracksIndexes.isEmpty()) {
+        if (trackIds.size() < m_sortedTracksIndexes.size()) {
```

Comment:
These assumptions are wrong. An arbitrary number of tracks could be added or removed. If the size didn't change then the number of additions equals the number of removals. That's all you could deduce.

Reply:
Making assumptions about such arbitrary preconditions is incorrect. The invocation of `select()` doesn't guarantee any of those.